inc/utils.php

```diff
@@ -1231,4 +1231,27 @@ function dsi_get_progetti_in_luogo($luogo_id)
     
         return $progetti;
     }
+}
+
+
+
+if(!function_exists("get_page_by_title_new")){
+	function get_page_by_title_new( $page_title, $output = OBJECT, $post_type = 'page' ) {
+		$query = new WP_Query([
+            'post_type' => 'page',
+            'title'     => $page_title,
+            'post_status' => 'publish', // Opzionale: solo pagine pubblicate
+            'posts_per_page' => 1, // Opzionale: limitare a un risultato
+        ]);
+        
+        $page = null;
+
+        if ($query->have_posts()) {
+            $query->the_post();
+            $page = get_post(); // Recupera l'oggetto del post corrente
+            wp_reset_postdata(); // Resetta i dati del loop di WordPress
+        }
+
+        return $page;
+	}
 }
```

Comment:
Suggerirei una formulazione un po' più compatta.

```suggestion
if (!function_exists("dsi_get_page_by_title")) {
    function dsi_get_page_by_title( $page_title ) {
        return (new WP_Query([
            'post_type' => 'page',
            'title'     => $page_title,
            'post_status' => 'publish',
            'posts_per_page' => 1,
        ]))->post;
    }
}
```

Reply:
NB: `get_page_by_title()` originale [non considera lo _status_](https://github.com/WordPress/WordPress/blob/0fc515b00579eb43c558b59ab73ecb18583429da/wp-includes/deprecated.php#L4590-L4595); se si vuole proprio replicarne il comportamento, sarebbe `'post_status' => 'any'`. 
Nel tema, è usata (solo) [all'attivazione](https://github.com/search?q=repo%3Aitalia%2Fdesign-scuole-wordpress-theme%20get_page_by_title&type=code); non è scontato cosa sia meglio.